### PR TITLE
Resolve #49 - Add ability to generate random Locale strings (e.g. en-US)

### LINKED ISSUE #49: ICU Locales
Please consider adding support for [ICU Locales](http://userguide.icu-project.org/locale)

Examples:
[`en_US`](http://www.localeplanet.com/icu/en-US/index.html)
[`en_GB`](http://www.localeplanet.com/icu/en-GB/index.html)
[`de_DE`](http://www.localeplanet.com/icu/de-DE/index.html)

### PR DESCRIPTION
**Pull request by n1try**
### Background
The PR attempts to resolve issue [49](https://github.com/Pallinder/go-randomdata/issues/49) by introducing a method to generate random [BCP 47](https://tools.ietf.org/html/bcp47)-compliant locale strings.

### Changes
- Add `randomdata.Locale()` method
- Add a sample of locale strings to `jsondata.go`
- Add tests

### Checklist
- [x] Tests added
- [x] Documentation updated to include changes (if needed)
- [x] Gofmt run on your code

### CODE REVIEW

File: jsondata.go [hunk 1]
```diff
@@ -2620,6 +2620,108 @@ var data = []byte(`{
         "W-SU",
         "Zulu"
     ],
+    "locales": [
```
Comment (ericadams) [inline jsondata.go]: this list appears to implement WordPress-specific locale codes as opposed to the normative standards put forth by IETF and W3 in 
http://tools.ietf.org/html/bcp47 
https://www.w3.org/TR/ltli/

Golang already implements an API for these tags in the experimental packages here:
https://godoc.org/golang.org/x/text/language

**Comment by n1try** [comment]
Please review again.

**Comment by Pallinder** [comment]
Yeah, missing a good explanation for this. I'm seeing the same issue on multiple PR:s though so I'm going to make sure it's documented.

**Comment by Pallinder** [comment]
I've added a PR template to follow. Please update the description of this PR to be inline with that template.

**Comment by n1try** [comment]
Done.

**Comment by Pallinder** [comment]
This is awesome. Merging this one and together with a few other PR:s we will have release 1.1.0. Great stuff everyone!

**Comment by ericadams** [comment]
well done @n1try!

**Pull request merged**